system/lib/pthread/library_pthread.c

```diff
@@ -907,13 +912,20 @@ int _emscripten_call_on_thread(
   q->calleeDelete = 1;
   // The called function will not be async if we are on the same thread; force
   // async if the user asked for that.
-  if (forceAsync) {
+  if (forceAsync == 1) {
     EM_ASM({
       setTimeout(function() {
         __emscripten_do_dispatch_to_thread($0, $1);
       }, 0);
     }, targetThread, q);
     return 0;
+  } else if (forceAsync == 2) {
```

Comment:
where does `2` come from?

Reply:
The header file declares three macros based in this function:
- `emscripten_dispatch_to_thread`: `forceAsync==0` Synchronous if in the same thread_id, otherwise asynchronous. 
- `emscripten_dispatch_to_thread_async`: `forceAsync==1` Force asynchronous call even if in the same thread_id.
- `emscripten_dispatch_to_thread_async`: `forceAsync==2` Always force a synchronous call.

We could change this parameter to `returnMode` or something to make it clearer.

Reply:
`mode` does sound good. or `dispatchMode`.

Reply:
Perfect, I will commit the change.